### DISCUSSION TITLE
Posibility of CSG modeling parameter change wrt time in a transient run

### DISCUSSION THREAD

**Discussion by WayneX-Wong** [neams-th-coe/cardinal#647] [Q&A] [answered]
Hi all,

I'm wondering if it's possible to change CSG modeling parameter in a single transient run using cardinal.

I'm not talking about on-line geometry modification caused by  unstructed mesh deformations, but artificial change of a parameter (such as different rod length Z in an OpenMC input of different time steps).

One way is to run one time step once, modify the OpenMC input after the end of the run, and then continue to run the next time step. But which seems very awkward. 

Does Cardinal have internal support for this situation?

**Comment by aprilnovak** [comment, marked as answer]
Hi @WayneX-Wong, great question. We don't currently have this implemented, but it would be straightforward to do so. After each run of OpenMC, you could change a surface/cell attribute (such as the radius of a cylinder) at the beginning of `OpenMCCellAverageProblem::setupProblem()`. For example, if you wanted to change the radius of a specific `XCylinder` which I have fetched and named `xcyl` locally:

```
void
OpenMCCellAverageProblem::setupProblem()
{
  xcyl.radius_ = 1.5; // change value however you like

  // ...
}
```

Cardinal will then re-init the geometry mapping on each time step if you set `fixed_mesh = false`, like this:

```
[Problem]
  type = OpenMCCellAverageProblem
  fixed_mesh = false
  ...
[]
```

In other words, if you can express how you want to change your surface parameters from C++, you should be able to do this from Cardinal. Designing this in a very general way may be challenging, but you could certainly add something like this.

**Reply by WayneX-Wong** [reply]
Thanks for your suggestion @aprilnovak , I'll try!
